packages/svelte/src/preprocessors.ts

```diff
@@ -99,3 +121,18 @@ function getBaseName(filename: string): string {
   const segments = filename.split('/');
   return segments[segments.length - 1].replace('.svelte', '');
 }
+
+function hasScriptTag(content: string): boolean {
+  // This regex is taken from the Svelte compiler code.
+  // They use this regex to find matching script tags that are passed to the `script` preprocessor hook:
+  // https://github.com/sveltejs/svelte/blob/bb83eddfc623437528f24e9fe210885b446e72fa/src/compiler/preprocess/index.ts#L144
+  // However, we remove the first part of the regex to not match HTML comments
+  const scriptTagRegex = /<script(\s[^]*?)?(?:>([^]*?)<\/script>|\/>)/gi;
+
+  // Regex testing is not a super safe way of checking for the presence of a <script> tag in the Svelte
```

Comment:
I agree this is a bit sketchy. Non blocking but have you looked at `import { parse } from 'svelte/compiler';` if it outputs something we could use that is a little bit more robust?

Reply:
I was just thinking about that too on my way to the train (apparently, walking really helps some time 😅). Will take a look and see if anything comes out of it

Reply:
Seems much more elegant and safer to use svelte's parser. Changed it in [31f1c52](https://github.com/getsentry/sentry-javascript/pull/5957/commits/31f1c520f59526a6a64bd44d9c431b252aa29bc5).

Reply:
Amazing. Super clean!